### PR TITLE
Fix+Test Timedelta.__mul__(nan)

### PR DESCRIPTION
**Pull request by jbrockmendel**
Specifically, `np.float64("NAN")` which in master gets caught by the `hasattr(other, "dtype")` branch and incorrectly returns `np.timedelta64("NaT", "ns")` instead of `NaT`

- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/scalar/timedelta/test_arithmetic.py [hunk 1]
```diff
@@ -328,6 +336,16 @@ def test_td_div_numeric_scalar(self):
         assert isinstance(result, Timedelta)
         assert result == Timedelta(days=2)
 
+    @pytest.mark.parametrize('nan', [np.nan, np.float64('NaN'), float('nan')])
```
Comment (jreback) [inline pandas/tests/scalar/timedelta/test_arithmetic.py]: nulls_fixture does this

File: pandas/tests/scalar/timedelta/test_arithmetic.py [hunk 2]
```diff
@@ -277,6 +277,14 @@ def test_td_mul_nat(self, op, td_nat):
         with pytest.raises(TypeError):
             op(td, td_nat)
 
+    @pytest.mark.parametrize('nan', [np.nan, np.float64('NaN'), float('nan')])
```
Comment (jreback) [inline pandas/tests/scalar/timedelta/test_arithmetic.py]: same

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/scalar/timedelta/test_arithmetic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/scalar/timedelta/test_arithmetic.py]: Yes, but it also includes several scalars we _dont_ want to include here.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**